pymatgen/io/lobster/lobsterenv.py

```diff
@@ -43,43 +43,42 @@ class LobsterNeighbors(NearNeighbors):
 
     def __init__(
         self,
-        are_coops=False,
-        filename_ICOHP=None,
-        valences=None,
-        limits=None,
-        structure=None,
-        additional_condition=0,
-        only_bonds_to=None,
-        perc_strength_ICOHP=0.15,
-        valences_from_charges=False,
-        filename_CHARGE=None,
-        which_charge="Mulliken",
-        adapt_extremum_to_add_cond=False,
-        add_additional_data_sg=False,
-        filename_blist_sg1=None,
-        filename_blist_sg2=None,
-        id_blist_sg1="ICOOP",
-        id_blist_sg2="ICOBI",
+        are_coops: bool = False,
+        filename_ICOHP: str = None,
+        valences: list[int | float] = None,
+        limits: tuple[float, float] | None = None,
+        structure: Structure = None,
```

Comment:
> @janosh, I tried to add more type hinting but mypy was raising very weird errors.

Is the `structure` arg really optional here? What happens if you don't supply a structure?

Reply:
You are correct. Same is true for "filename_ICOHP".